FinHubSA/information-access
# feat: 138 search by title author journal

# Pull Request

## Scope
https://dev.azure.com/JHNMAT015/ECO5037S%20Front%20end/_workitems/edit/138

## Work Done
Configured vuex to get from new api endpoints and implemented searchbar functionality. Now possible to search by author, title or journal name by selecting a particular field to search. Routing between pages through search bar implemented, both search icon and enter will initiate a search. Removal of navbar item for search results page.

Connected end points:
http://localhost:5000/api/articles/title?title=inc  (searching titles containing 'inc')
http://localhost:5000/api/articles/journal?journalName=econ (searching journal names containing 'econ')
http://localhost:5000/api/articles/author?AuthorSurname=ch (searching author surnames containing 'ch')

Note: currently running on local using develop branch of information storage

## Steps to Test

```npm run start```
Walk through site from home to search results using search bar and test different kinds of user behavior. Check accuracy of search results by comparing with api output eg: if search for title with 'inc' check http://localhost:5000/api/articles/title?title=inc

## Screenshot
![image](https://user-images.githubusercontent.com/80747408/136474339-dae29de1-0021-4f96-82a2-9b1635f48902.png)
![image](https://user-images.githubusercontent.com/80747408/136474562-fbd30e88-a11f-4b3c-9e90-8b3f8000bb40.png)
